ironfish/src/rpc/routes/wallet/getNotes.ts

```diff
@@ -0,0 +1,116 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+import * as yup from 'yup'
+import { BufferUtils, CurrencyUtils } from '../../../utils'
+import { AssetValue } from '../../../wallet/walletdb/assetValue'
+import { DecryptedNoteValue } from '../../../wallet/walletdb/decryptedNoteValue'
+import { ApiNamespace, router } from '../router'
+import { getAccount } from './utils'
+
+export type GetNotesRequest = {
+  account?: string
+  pageSize?: number
+  pageToken?: string
+}
+
+export type RpcWalletNote = {
+  value: string
+  assetId: string
+  assetName: string
+  memo: string
+  sender: string
+  noteHash: string
+  transactionHash: string
+  index: number | null
+  spent: boolean | undefined
+}
+
+export type GetNotesResponse = {
+  notes: Array<RpcWalletNote>
+  nextPageToken?: string
+}
+
+export const GetNotesRequestSchema: yup.ObjectSchema<GetNotesRequest> = yup
+  .object({
+    account: yup.string().trim(),
+    pageSize: yup.number().min(1),
+    pageToken: yup.string(),
+  })
+  .defined()
+
+export const GetNotesResponseSchema: yup.ObjectSchema<GetNotesResponse> = yup
+  .object({
+    notes: yup
+      .array(
+        yup
+          .object({
+            value: yup.string().defined(),
+            assetId: yup.string().defined(),
+            assetName: yup.string().defined(),
+            memo: yup.string().trim().defined(),
+            sender: yup.string().defined(),
+            noteHash: yup.string().defined(),
+            transactionHash: yup.string().defined(),
+            index: yup.number(),
+            spent: yup.boolean(),
+          })
+          .defined(),
+      )
+      .defined(),
+    nextPageToken: yup.string(),
+  })
+  .defined()
+
+router.register<typeof GetNotesRequestSchema, GetNotesResponse>(
+  `${ApiNamespace.wallet}/getNotes`,
+  GetNotesRequestSchema,
+  async (request, node): Promise<void> => {
+    const account = getAccount(node, request.data.account)
+    const pageSize = request.data.pageSize ?? 100
```

Comment:
thoughts on implementing a max page size?

Reply:
That's a good idea. 100 was an arbitrary default choice 😅 

Some max/default size based on the size of the response would be better

Reply:
Thinking about this a bit more and I worry about setting a max page size that's too small (or too big) for integration developers.

I think that's something that I would leave up to the node runner to decide, but we don't have a pattern yet for per-endpoint configuration.

Reply:
sgtm